### PR TITLE
fix: Add unit tests for math utility functions and fix median calculation

### PR DESCRIPTION
**Pull request by tiendn**
### Description
This pull request introduces the following changes to the math utility functions:
#### Fix Median Calculation:
Corrected the median function to handle even-length arrays properly by averaging the two middle numbers.

#### Add Unit Tests:
Implemented unit tests for the math utility functions using Chai to ensure their correctness and reliability.
The functions tested include:
- `median`: Tests for both odd and even-length arrays.
- `sum`: Verifies the sum of an array of numbers.
- `mean`: Checks the calculation of the mean.
- `stdDev`: Validates the standard deviation calculation, including cases with negative numbers.
- `randomInt`: Ensures the generated random integer falls within the specified range.

### Drive-by changes
- Code: Updated the median function in math.ts to correctly calculate the median for even-length arrays.
- Tests: Added a new test file math.test.ts with comprehensive test cases for each function.

### Related issues

Fixes [#[4754]](https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/4754)

### Backward compatibility
Yes

### Testing

- All tests have been executed and passed successfully, confirming the correctness of the functions.
- Special attention was given to edge cases, such as arrays with negative numbers and random integer generation.

### CODE REVIEW

**Review by jmrossy — APPROVED** [review]
Good catch, thanks!

**Comment by jmrossy** [comment]
@tiendn merge in the latest from main to fix the CI failure with the CLI test

File: typescript/utils/src/math.test.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { expect } from 'chai';
+
+import { mean, median, randomInt, stdDev, sum } from '../src/math.js';
```
Comment (jmrossy) [inline typescript/utils/src/math.test.ts:3]: Sorry I missed this in my first pass. This import path should be just `'./math.js'`

Comment (tiendn) [reply typescript/utils/src/math.test.ts]: done

**Comment by jmrossy** [comment]
@tiendn your new code hasn't been formatted correctly. Run `yarn prettier` to fix

**Comment by tiendn** [comment]
I've updated. Help me again. Tks

**Pull request merged**